src/Aspire.Hosting.Azure.AppService/AzureAppServiceWebsiteContext.cs

```diff
@@ -0,0 +1,346 @@
+        }
+
+        if (value is IAzureKeyVaultSecretReference vaultSecretReference)
+        {
+            if (parent is null)
+            {
+                return (AllocateKeyVaultSecretUriReference(vaultSecretReference), SecretType.KeyVault);
+            }
+
+            return (AllocateParameter(vaultSecretReference, secretType: SecretType.KeyVault), SecretType.KeyVault);
+        }
+
+        if (value is EndpointReferenceExpression epExpr)
+        {
+            var context = environmentContext.GetAppServiceContext(epExpr.Endpoint.Resource);
+            var mapping = context._endpointMapping[epExpr.Endpoint.EndpointName];
+            var val = GetValue(mapping, epExpr.Property);
+            return (val, secretType);
+        }
+
+        if (value is ReferenceExpression expr)
+        {
+            if (expr.Format == "{0}" && expr.ValueProviders.Count == 1)
+            {
+                return ProcessValue(expr.ValueProviders[0], secretType, parent);
+            }
+
+            var args = new object[expr.ValueProviders.Count];
+            var index = 0;
+            var finalSecretType = SecretType.None;
+
+            foreach (var vp in expr.ValueProviders)
+            {
+                var (val, secret) = ProcessValue(vp, secretType, expr);
+                if (secret != SecretType.None)
+                {
+                    finalSecretType = SecretType.Normal;
+                }
+                args[index++] = val;
+            }
+
+            return (new BicepFormatString(expr.Format, args), finalSecretType);
+        }
+
+        throw new NotSupportedException($"Unsupported value type {value.GetType()}");
+    }
+
+    private static BicepValue<string> ResolveValue(object val)
+    {
+        return val switch
+        {
+            BicepValue<string> s => s,
+            string s => s,
+            ProvisioningParameter p => p,
+            BicepFormatString fs => BicepFunction2.Interpolate(fs),
+            _ => throw new NotSupportedException($"Unsupported value type {val.GetType()}")
+        };
+    }
+
+    public void BuildWebSite(AzureResourceInfrastructure infra)
+    {
+        _infrastructure = infra;
+
+        // We need to reference the container registry URL so that it exists in the manifest
+        var containerRegistryUrl = environmentContext.Environment.ContainerRegistryUrl.AsProvisioningParameter(infra);
+        var appServicePlanParameter = environmentContext.Environment.PlanIdOutputReference.AsProvisioningParameter(infra);
+        var acrMidParameter = environmentContext.Environment.ContainerRegistryManagedIdentityId.AsProvisioningParameter(infra);
+        var acrClientIdParameter = environmentContext.Environment.ContainerRegistryClientId.AsProvisioningParameter(infra);
+        var containerImage = AllocateParameter(ResourceExpression.GetContainerImageExpression(Resource));
+
+        var webSite = new WebSite("webapp")
```

Comment:
(nit) why the hard-coded "webapp" instead of being based on the Resource name?

Reply:
Does it matter? It's in a module.

Reply:
It doesn't. Just in consistent with ACA.